auth.py

```
@@ -0,0 +1,200 @@
+    ) -> None:
+        """Initialize the auth client."""
+        self.session = session
+        self.host = host
+        self.base_url = f"http://{host}"
+        self.session_id = None
+        
+    async def authenticate_with_access_code(self, access_code: str) -> str | None:
+        """Authenticate using Device Access Code and return session ID."""
+        try:
+            # Step 1: Get the login page to extract any necessary tokens/nonces
+            login_url = f"{self.base_url}/cgi-bin/login.ha"
```

Ruff (0.12.2)

Local variable `login_url` is assigned to but never used

Remove assignment to unused variable `login_url`

(F841)